### ISSUE TITLE
Lucene errors while doing index of source tree

### ISSUE THREAD

**Issue by ilovemaui** [oracle/opengrok#2122]
running on Redhat 6.8
OpenGrok Version 1.1-RC26
Java - jdk1.8.0_72
apache-tomcat 8.5.4
While performing this index, I saw a lot of severe errors trying to index my source tree.

Here are some examples:
SEVERE: ERROR addFile(): /myopengrok/myfile.xml
org.apache.lucene.store.AlreadyClosedException: this IndexWriter is closed
        at org.apache.lucene.index.IndexWriter.ensureOpen(IndexWriter.java:897)
        at org.apache.lucene.index.IndexWriter.ensureOpen(IndexWriter.java:911)
        at org.apache.lucene.index.IndexWriter.updateDocument(IndexWriter.java:1725)
        at org.apache.lucene.index.IndexWriter.addDocument(IndexWriter.java:1464)
        at org.opensolaris.opengrok.index.IndexDatabase.addFile(IndexDatabase.java:749)
        at org.opensolaris.opengrok.index.IndexDatabase.lambda$null$1(IndexDatabase.java:1075)
        at java.util.stream.Collectors.lambda$groupingByConcurrent$51(Collectors.java:1070)
        at java.util.stream.ReferencePipeline.lambda$collect$1(ReferencePipeline.java:496)
        at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
        at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1374)
        at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
        at java.util.stream.ForEachOps$ForEachTask.compute(ForEachOps.java:291)
        at java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:731)
        at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
        at java.util.concurrent.ForkJoinPool$WorkQueue.execLocalTasks(ForkJoinPool.java:1040)
        at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1058)
        at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
        at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
Caused by: java.lang.ArrayIndexOutOfBoundsException: -65536
        at org.apache.lucene.index.TermsHashPerField.writeByte(TermsHashPerField.java:198)
        at org.apache.lucene.index.TermsHashPerField.writeVInt(TermsHashPerField.java:224)
        at org.apache.lucene.index.FreqProxTermsWriterPerField.writeProx(FreqProxTermsWriterPerField.java:80)
        at org.apache.lucene.index.FreqProxTermsWriterPerField.addTerm(FreqProxTermsWriterPerField.java:184)
        at org.apache.lucene.index.TermsHashPerField.add(TermsHashPerField.java:185)
        at org.apache.lucene.index.DefaultIndexingChain$PerField.invert(DefaultIndexingChain.java:786)
        at org.apache.lucene.index.DefaultIndexingChain.processField(DefaultIndexingChain.java:430)
        at org.apache.lucene.index.DefaultIndexingChain.processDocument(DefaultIndexingChain.java:392)
        at org.apache.lucene.index.DocumentsWriterPerThread.updateDocument(DocumentsWriterPerThread.java:240)
        at org.apache.lucene.index.DocumentsWriter.updateDocument(DocumentsWriter.java:497)
        at org.apache.lucene.index.IndexWriter.updateDocument(IndexWriter.java:1729)
        at org.apache.lucene.index.IndexWriter.addDocument(IndexWriter.java:1464)
        at org.opensolaris.opengrok.index.IndexDatabase.addFile(IndexDatabase.java:749)
        at org.opensolaris.opengrok.index.IndexDatabase.lambda$null$1(IndexDatabase.java:1075)
        at java.util.stream.Collectors.lambda$groupingByConcurrent$51(Collectors.java:1070)
        at java.util.stream.ReferencePipeline.lambda$collect$1(ReferencePipeline.java:496)
        at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
        at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1374)
        at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
        at java.util.stream.ForEachOps$ForEachTask.compute(ForEachOps.java:291)
        at java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:731)
        at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
        at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
        ... 2 more

May 17, 2018 12:20:54 PM org.opensolaris.opengrok.index.IndexDatabase indexParallel
WARNING: 71529 failures (12.3%) while parallel-indexing
May 17, 2018 12:20:54 PM org.opensolaris.opengrok.index.IndexDatabase update
SEVERE: Failed with unexpected RuntimeException
org.apache.lucene.store.AlreadyClosedException: count=36

The other errors seem the same - but we don't get this last error with the count=36. 

I hope this is enough info to go with to determine error. thanks.

**Comment by idodeclare** [comment]
Alas @ilovemaui we may need more detail to debug. Are you certain that excerpt is the original exception causing the `close()`?

The problem may be content-specific; can you determine a specific file and make it public?

**Comment by IamTHEvilONE** [comment]
I'm seeing this as well on opengrok-index on 1.2.14.

2019-06-19 13:20:11.386-0400 SEVERE t12396 IndexDatabase.lambda$indexParallel$1: ERROR addFile(): /path/to/my/file.cpp
org.apache.lucene.store.AlreadyClosedException: this IndexWriter is closed
        at org.apache.lucene.index.IndexWriter.ensureOpen(IndexWriter.java:680)
        at org.apache.lucene.index.IndexWriter.ensureOpen(IndexWriter.java:694)
        at org.apache.lucene.index.IndexWriter.updateDocument(IndexWriter.java:1613)
        at org.apache.lucene.index.IndexWriter.addDocument(IndexWriter.java:1235)
        at org.opengrok.indexer.index.IndexDatabase.addFile(IndexDatabase.java:780)
        at org.opengrok.indexer.index.IndexDatabase.lambda$indexParallel$1(IndexDatabase.java:1193)
        at java.util.stream.Collectors.lambda$groupingByConcurrent$51(Collectors.java:1070)
        at java.util.stream.ReferencePipeline.lambda$collect$1(ReferencePipeline.java:496)
        at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
        at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1382)
        at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482)
        at java.util.stream.ForEachOps$ForEachTask.compute(ForEachOps.java:291)
        at java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:731)
        at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
        at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
        at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
        at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)

I have several projects indexed by OpenGrok ... to give some examples:

opengrok0.0.log:2019-06-19 13:20:22.156-0400 WARNING t10962 IndexDatabase.indexParallel: 17089 failures (100.0%) while parallel-indexing
opengrok0.0.log:2019-06-19 13:20:44.333-0400 WARNING t10960 IndexDatabase.indexParallel: 16715 failures (100.0%) while parallel-indexing
opengrok0.0.log:2019-06-19 13:20:49.124-0400 WARNING t10963 IndexDatabase.indexParallel: 16121 failures (100.0%) while parallel-indexing
opengrok0.0.log:2019-06-19 13:21:06.537-0400 WARNING t10961 IndexDatabase.indexParallel: 21120 failures (100.0%) while parallel-indexing

My current indexing command is:

opengrok-indexer -J=-Djava.util.logging.config.file=/usr/opengrok/doc/logging.properties \
        -J=-XX:+HeapDumpOnOutOfMemoryError \
        -J=-XX:HeapDumpPath=/usr/opengrok/heap_dumps \
        -a /usr/opengrok/lib/opengrok.jar -- \
        -c /usr/local/bin/ctags \
        -s /usr/opengrok/src -d /usr/opengrok/data -H -P -S -G \
        -W /usr/opengrok/etc/configuration.xml -U http://localhost:8080/source/

I've modified the index logging.properties to capture 10 log files instead of 3, as well as running in a screen session to capture the standard out of the opengrok-index command.

In terms of similar/related errors, this is what I see:

2019-06-19 14:24:07.532-0400 SEVERE t10961 IndexDatabase.indexParallel: 4342 successes (45.0%) after aborting parallel-indexing
java.util.concurrent.ExecutionException: java.lang.NoSuchMethodError
        at java.util.concurrent.ForkJoinTask.get(ForkJoinTask.java:1006)
        at org.opengrok.indexer.index.IndexDatabase.indexParallel(IndexDatabase.java:1229)
        at org.opengrok.indexer.index.IndexDatabase.update(IndexDatabase.java:496)
        at org.opengrok.indexer.index.IndexDatabase$1.run(IndexDatabase.java:225)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.NoSuchMethodError
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
        at java.util.concurrent.ForkJoinTask.getThrowableException(ForkJoinTask.java:598)
        at java.util.concurrent.ForkJoinTask.get(ForkJoinTask.java:1005)
        ... 8 more
Caused by: java.lang.NoSuchMethodError
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
        at java.util.concurrent.ForkJoinTask.getThrowableException(ForkJoinTask.java:598)
        at java.util.concurrent.ForkJoinTask.reportException(ForkJoinTask.java:677)
        at java.util.concurrent.ForkJoinTask.invoke(ForkJoinTask.java:735)
        at java.util.stream.ForEachOps$ForEachOp.evaluateParallel(ForEachOps.java:160)
        at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateParallel(ForEachOps.java:174)
        at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:233)
        at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
        at java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:583)
        at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:496)
        at org.opengrok.indexer.index.IndexDatabase.lambda$indexParallel$2(IndexDatabase.java:1182)
        at java.util.concurrent.ForkJoinTask$AdaptedCallable.exec(ForkJoinTask.java:1424)
        at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
        at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
        at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
        at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
Caused by: java.lang.NoSuchMethodError: java.nio.MappedByteBuffer.position(I)Ljava/nio/MappedByteBuffer;
        at org.opengrok.indexer.analysis.executables.ELFAnalyzer$ELFHeader.<init>(ELFAnalyzer.java:229)
        at org.opengrok.indexer.analysis.executables.ELFAnalyzer.parseELF(ELFAnalyzer.java:108)
        at org.opengrok.indexer.analysis.executables.ELFAnalyzer.analyze(ELFAnalyzer.java:93)
        at org.opengrok.indexer.analysis.AnalyzerGuru.populateDocument(AnalyzerGuru.java:586)
        at org.opengrok.indexer.index.IndexDatabase.addFile(IndexDatabase.java:754)
        at org.opengrok.indexer.index.IndexDatabase.lambda$indexParallel$1(IndexDatabase.java:1193)
        at java.util.stream.Collectors.lambda$groupingByConcurrent$51(Collectors.java:1070)
        at java.util.stream.ReferencePipeline.lambda$collect$1(ReferencePipeline.java:496)
        at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
        at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1382)
        at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482)
        at java.util.stream.ForEachOps$ForEachTask.compute(ForEachOps.java:291)
        at java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:731)
        at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
        at java.util.concurrent.ForkJoinPool$WorkQueue.execLocalTasks(ForkJoinPool.java:1040)
        at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1058)
        ... 2 more

**Comment by vladak** [comment]
The `NoSuchMethodError` in ELF analyzer is already tracked by #2815.

**Comment by Ymoise** [comment]
I'm running in a container based on a docker image built from opengrok/docker, and an altered index.sh that looks like this:

#!/bin/bash

LOCKFILE=/var/run/opengrok-indexer
URI="http://localhost:8080"
# $OPS can be overwritten by environment variable
OPS=${INDEXER_FLAGS:='-H -P -G'}

if [ -f "$LOCKFILE" ]; then
        date +"%F %T Indexer still locked, skipping indexing"
        exit 1
fi

touch $LOCKFILE

date +"%F %T Indexing starting"
opengrok-indexer \
    -J=-Xmx8g \
    -J=-d64 \
    -J=-server \
    -a /opengrok/lib/opengrok.jar -- \
    -m 256 \
    -v \
    --repository /opengrok/src/master \
    -s /opengrok/src \
    -d /opengrok/data \
    --remote off \
    -W /opengrok/etc/configuration.xml \
    -U "$URI" \
    $OPS \
    $INDEXER_OPT "$@"
date +"%F %T Indexing finished"

---

Initial indexing ends up with lots of these errors:

Mar 12, 2020 9:28:27 AM org.opengrok.indexer.index.IndexDatabase lambda$indexParallel$2
SEVERE: ERROR addFile(): /opengrok/src/main/Qmatlab_util/source/tool/aera/Algorithms/CNFs/ADCProject/Hires2AerialSim/findDefRrefCADshift.m
org.apache.lucene.store.AlreadyClosedException: this IndexWriter is closed
        at org.apache.lucene.index.IndexWriter.ensureOpen(IndexWriter.java:681)
        at org.apache.lucene.index.IndexWriter.ensureOpen(IndexWriter.java:695)
        at org.apache.lucene.index.IndexWriter.updateDocument(IndexWriter.java:1591)
        at org.apache.lucene.index.IndexWriter.addDocument(IndexWriter.java:1213)
        at org.opengrok.indexer.index.IndexDatabase.addFile(IndexDatabase.java:774)
        at org.opengrok.indexer.index.IndexDatabase.lambda$indexParallel$2(IndexDatabase.java:1271)
        at java.util.stream.Collectors.lambda$groupingByConcurrent$51(Collectors.java:1070)
        at java.util.stream.ReferencePipeline.lambda$collect$1(ReferencePipeline.java:496)
        at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
        at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1382)
        at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482)
        at java.util.stream.ForEachOps$ForEachTask.compute(ForEachOps.java:291)
        at java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:731)
        at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
        at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
        at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
        at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
Caused by: java.lang.ArrayIndexOutOfBoundsException: -65536
        at org.apache.lucene.index.TermsHashPerField.writeByte(TermsHashPerField.java:198)
        at org.apache.lucene.index.TermsHashPerField.writeVInt(TermsHashPerField.java:221)
        at org.apache.lucene.index.FreqProxTermsWriterPerField.writeProx(FreqProxTermsWriterPerField.java:80)
        at org.apache.lucene.index.FreqProxTermsWriterPerField.newTerm(FreqProxTermsWriterPerField.java:121)
        at org.apache.lucene.index.TermsHashPerField.add(TermsHashPerField.java:178)
        at org.apache.lucene.index.DefaultIndexingChain$PerField.invert(DefaultIndexingChain.java:862)
        at org.apache.lucene.index.DefaultIndexingChain.processField(DefaultIndexingChain.java:442)
        at org.apache.lucene.index.DefaultIndexingChain.processDocument(DefaultIndexingChain.java:406)
        at org.apache.lucene.index.DocumentsWriterPerThread.updateDocument(DocumentsWriterPerThread.java:250)
        at org.apache.lucene.index.DocumentsWriter.updateDocument(DocumentsWriter.java:495)
        at org.apache.lucene.index.IndexWriter.updateDocument(IndexWriter.java:1594)
        ... 14 more

----

I initially thought it was user error causing this, because the original index.sh used for the image was based on a 14 month old index.sh file (that was why it was asking to save the config in the wrong place), but having updated the index.sh... it's still doing it.

Would appreciate help.

**Comment by Ymoise** [comment]
Tried removing the memory related flags from index.sh:

---

#!/bin/bash

LOCKFILE=/var/run/opengrok-indexer
URI="http://localhost:8080"
# $OPS can be overwritten by environment variable
OPS=${INDEXER_FLAGS:='-H -P -G'}

if [ -f "$LOCKFILE" ]; then
        date +"%F %T Indexer still locked, skipping indexing"
        exit 1
fi

touch $LOCKFILE

date +"%F %T Indexing starting"
opengrok-indexer \
    -a /opengrok/lib/opengrok.jar -- \
    -v \
    --repository /opengrok/src/main \
    -s /opengrok/src \
    -d /opengrok/data \
    --remote off \
    -W /opengrok/etc/configuration.xml \
    -U "$URI" \
    $OPS \
    $INDEXER_OPT "$@"
date +"%F %T Indexing finished"

rm -f $LOCKFILE

---

Still getting the same SEVEREs.

**Comment by Ymoise** [comment]
I have one container, created EXACTLY the same way, and it works perfectly... and yet, every single one I create, other than that one, miraculous first seem to go sour. 

I tried fiddling with the threads flag, reducing it to less than the max CPUs, in case reaching the CPU load limit was causing this issue (it was getting pretty close in the runs I tried).  

I tried fiddling with the memory and heap flags, in case the tuning instructions for large datasets were no longer needed. 

I tried it on a dataset that contained multiple projects. 

I tried it on one that only had ONE project. 

All produced the same SEVERE and issue.

**Comment by Ymoise** [comment]
Ok, could this issue be in any way caused, not by settings, but by the data that's being indexed?

My team works on separate branches. What I'm doing is I'm cloning the separate branches into separate source folders which serve as root folders for separate containers, each running opengrok, so that each branch can be accessed and searched separately. 

When I use the 1.5 year old opengrok image, it takes forever and sometimes sticks for no reason, so I can't reliably update the containers automatically, but I can generally update any of them I want.

However, if I use the latest image, it's reliable, and indexed blisteringly fast... but fails in all versions except one!

I just cloned the same branch to a different source folder, set up a container for it, ran the index, and it WORKED, but any other branch I try, it just doesn't work.

And I don't get why. It's cloned already, so you can't even say that there might be connection issues, or permissions at the remote end or whatever. 

One branch, when cloned, can be indexed reliably, quickly, and most importantly SUCCESSFULLY, but any other branch I try it on, just fails with the severes I mentioned. 

One of them even failed within 10 minutes of starting. 

I really am running out of ideas here... :(

**Comment by idodeclare** [comment]
@Ymoise , it does have the same hint that it's content-specific since the stack trace seems equivalent to the original submission by @ilovemaui.

Per your log excerpt, what is the file type of your _/opengrok/src/files_? What is its size? Could you summarize its structure or possibly share its contents?

But OpenGrok indexes in parallel, so the file above might not be the root problem. Is that the only `SEVERE` event in the logs? Are there other files throwing exceptions? Are they similarly "Caused by: `java.lang.ArrayIndexOutOfBoundsException`: -65536"? Anything remarkable about any other affected files?

**Comment by idodeclare** [comment]
Also in studying some more, I see some Lucene discussion that `ArrayIndexOutOfBoundsException` might be raised when dialing up RAM buffer size (`-m 256`), as Lucene exhausts its buffers (but without a more intuitive `OutOfMemoryError`).

I see you removed all memory-related switches. Any chance you're now seeing `OutOfMemoryError` instead?

**Comment by Ymoise** [comment]
I'm running with the first version, the one with the memory switches, not the second one, without them. 

The version with the memory switches removed was a test to see if that would help, but I'm afraid it didn't. 

I looked into the files at the point of it falling apart and found that the culprit is a 1.1gb matlab file that exists in most branches, but not the one that works.

Presumably, that's too much for it to chew. 

Ok, what do I do, now? I can't get rid of the file. 

Is there any way I could adjust opengrok to handle it?

**Comment by idodeclare** [comment]
It's probably worth trying with high heap (`-J=-Xmx8g` or more) and standard RAM buffer size (~~`-m 256`~~ or `-m 16`).

If acceptable, you could try to ignore files with:

```
  -i, --ignore pattern
	Ignore matching files (prefixed with 'f:' or no prefix) or directories
	(prefixed with 'd:'). Pattern supports wildcards (example: -i '*.so'
	-i d:'test*'). Option may be repeated.
```

(BTW I wonder what does `-J=-d64` do? I don't recognize that switch. When I try it on a test JVM, I see an error:

```
$ java -d64
Unrecognized option: -d64
Error: Could not create the Java Virtual Machine.
Error: A fatal exception has occurred. Program will exit.
```
)

**Comment by Ymoise** [comment]
Thanks! I got that error once, too, but I figured it was a freak occurrence, because when I recreated the container, it didn't happen again. Now that you mentioned it, I went to look for it. 

Apparently, it's a flag that's only useful in Solaris, but useless on Linux and Windows. On Solaris, it's used to specify whether you want to run Java in 64-bit or 32-bit (source: https://stackoverflow.com/questions/31078803/usage-of-d32-and-d64-while-launching-java/32007151). 

My predecessor at this project, who started it 1.5 years ago and who set up the basics, got the flag from the opengrok wiki (https://github.com/oracle/opengrok/wiki/Tuning-for-large-code-bases). 

I _was_ wondering, while I was working at this problem, whether that page might need updating. I guess it does, after all. :)

---

Anyway, I'll try the standard buffer (16) with the current heap size (8g) and we'll see if that helps.

If push comes to shove... I'll ignore the file and pray they don't give it some siblings in the future. 

Will let you know how it went.

Thanks ever so much! :)

**Comment by Ymoise** [comment]
The buffer adjustment didn't work. 

Ended up ignoring the file, and the indexing went through. 

It's not a solution by any measure, but at least I can finally start adding more branches' indexes.

---

It's weird, how the 1.5 year old version of opengrok can handle the file, but the latest can't.

Anyway - One last question: Looking at the containers, they're all "622GB used", regardless of the size of our data (which I can tell you is definitely not 622GB). Is this normal for OpenGrok containers?

**Comment by vladak** [comment]
> Anyway - One last question: Looking at the containers, they're all "622GB used", regardless of the size of our data (which I can tell you is definitely not 622GB). Is this normal for OpenGrok containers?

You can see the breakdown of image size on https://microbadger.com/images/opengrok/docker , currently it says 498.9 MB. Where exactly do you see the 622 GB value ?

**Comment by Ymoise** [comment]
I got it from the "df -h" command.

But, considering I just updated one of my containers and ALL four of the containers now register as having grown... I think the output is for all containers, put together.

I must have misunderstood it, then. 

My apologies. I'm a novice at Linux, I'm afraid.

**Comment by vladak** [comment]
> (BTW I wonder what does `-J=-d64` do? I don't recognize that switch. When I try it on a test JVM, I see an error:

Interesting, I did not know this is Solaris specific. I adjusted the wiki.

**Comment by idodeclare** [comment]
@vladak , there's one more use of `-d64` on that wiki for the sample opengrok-indexer command.

**Comment by idodeclare** [comment]
> It's not a solution by any measure...

You're unblocked, so isn't it at least a wee bit of a solution. 🧐

> It's weird, how the 1.5 year old version of opengrok can handle the file, but the latest can't.

Well, we add analyzers and change behavior for various reasons. What analyzer did OpenGrok pick for that file? What's the file name?
